pallets/liquid-staking-v2/src/lib.rs

```diff
@@ -216,8 +204,57 @@ mod pallet {
         }
     }
 
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        /// Try to pay off over the `UnstakeQueue` while blockchain is on idle.
+        ///
+        /// It breaks when:
+        ///     - Pallet's balance is insufficiant.
+        ///     - Queue is empty.
+        ///     - `remaining_weight` is less than one pop_queue needed.
+        fn on_idle(_n: BlockNumberFor<T>, remaining_weight: Weight) -> Weight {
+            let mut remaining_weight = remaining_weight;
+            let base_weight = T::WeightInfo::pop_queue();
+            loop {
+                // Check weight is enough
+                if remaining_weight < base_weight {
+                    break;
+                }
+
+                if Self::unstake_queue().is_empty() {
+                    break;
+                }
+
+                let (who, amount) = Self::unstake_queue()
+                    .first()
+                    .expect("Has been checked before.")
+                    .clone();
+
+                if T::Currency::transfer(
+                    T::StakingCurrency::get(),
+                    &Self::account_id(),
+                    &who,
+                    amount,
+                )
+                .is_err()
+                {
+                    break;
+                }
+
+                remaining_weight -= base_weight;
+                UnstakeQueue::<T>::mutate(|v| {
```

Comment:
can we make this a function  to make it more meaningful

Reply:
I could add some more comments.